.github/workflows/create-release.yml

```
@@ -27,10 +27,23 @@ jobs:
           python-version: 3.8
       - name: Change version number
         id: version
+        # See this issue for explanation and testing:
+        # https://github.com/cmu-delphi/delphi-epidata/pull/1473
         run: |
-          python -m pip install bump2version
-          echo -n "::set-output name=next_tag::"
-          bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,
+          python -m pip install bump-my-version
+          echo -n "next_tag=" >> $GITHUB_OUTPUT
+          allowed_pattern="^(major|minor|patch|[0-9]+\.[0-9]+\.[0-9]+)$"
+
+          if [[ ! ${{ github.event.inputs.versionName }} =~ $allowed_pattern ]]; then
+            echo "\nInvalid version name: ${{ github.event.inputs.versionName }}"
+            exit 1
+          fi
+
+          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "${{ github.event.inputs.versionName }}"
+          else
+            echo $(bump-my-version show-bump | grep ${{ github.event.inputs.versionName }} | sed -r s,"^.*─ ",,)
+          fi
```

`bump-my-version` doesnt look a whole lot better than `bump2version`...  what if we just keep using bump2 but with the logic to either extract a numeric version from the input w/ your regex, or otherwise try to generate a new numeric version?

```suggestion
          python -m pip install bump2version
          if [[ ${{ github.event.inputs.versionName }} =~ "^[0-9]+\.[0-9]+\.[0-9]+$" ]]; then
            NEXT_TAG="${{ github.event.inputs.versionName }}"
          else
            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
          fi
          bump2version --new-version $NEXT_TAG _ignored_arg_
          echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
```

and then below:
`bump2version --allow-dirty --new-version ${{ steps.version.outputs.next_tag }} _ignored_arg_`

---

Sure, I pushed a fix.